src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -114,6 +115,9 @@ export class RoslynLanguageServer {
      */
     private _solutionFile: vscode.Uri | undefined;
 
+    /** The project files previously opened; we hold onto this for the same reason as _solutionFile. */
+    private _projectFiles: vscode.Uri[] = new Array<vscode.Uri>();
```

Comment:
I understand the solution file is also a moving target to cache. But the contents were cache agnostic. The project file list on the other hand as a cache seems more volatile if cache and not  tracked continuously.

Reply:
Not a blocker, just want to get on the same  page

Reply:
Array constructor could have been monkey patched. Prefer `[]` since it is guaranteed to construct an array.

```suggestion
    private _projectFiles: vscode.Uri[] = [];
```